### DISCUSSION TITLE
Why DeliverToStorage creates directory in Azure StorageContainers with "preview" tag?

### DISCUSSION THREAD

**Discussion by HazardMK** [microsoft/AL-Go#1220] [Q&A] [answered]
Hi,

When DeliverToStorage is enabled for repository it delivers artifacts after each successful build to Azure Storage Containers. Everything works fine, we can find the zipped .app files per repository in separate containers. Directories are created by version, but for some reason it also adds "preview" at the end of the directory name. Is there a way to change/disable this "preview" tag and what's the meaning behind this? Also does it mean that in folder named "preview" without any version we can find the latest build version?

![image](https://github.com/user-attachments/assets/e20feee9-021c-46cc-82f5-4cd6adaa91d1)

**Comment by freddydk** [comment, marked as answer]
Correct - after every build (if continuousDelivery is enabled), the apps in the preview container will be updated.
Every build will be added to it's own folder called version-preview (if continuousDelivery is enabled)

Upon release, you will have a container with the version number and another container called latest with the latest version

only way to avoid the preview folders is to disable continuousdelivery and only deliver on release.

**Reply by HazardMK** [reply]
Hi Freddy,

Thanks for your reply. Indeed upon releasing (using actions) it creates a "latest" folder and a folder with a version without "preview". Is there a way to add a version to a .zip filename when delivering to az storage?

**Reply by freddydk** [reply]
In the StorageContext secret, you can setup the container name and the blob name, where certain placeholders gets replaced with the real value. Read more here: https://aka.ms/algosecrets#storagecontext
